### PR TITLE
feat(widget): added success and failure redirect urls to widget localStorage for provider connections

### PR DESCRIPTION
**Pull request by RamilGaripov**
refs. metriport/metriport#711

### Description

- Added the query parameters `redirectUrl` and `failRedirectUrl` to the Widget, allowing the CX to redirect their users to their custom urls based on success/failure during provider connection in the widget.

### Screenshot
Screenshot 2023-07-27 at 12 28 20 PM

### CODE REVIEW

File: packages/connect-widget/src/shared/util.ts [hunk 1]
```diff
@@ -17,6 +17,15 @@ export function redirectToMain(navigate: NavigateFunction, searchParams: URLSear
   }
 }
 
+// redirects to the custom redirect url
+export function redirectToCustomUrl(url: string) {
+  try {
+    window.location.href = url;
+  } catch (err) {
+    capture.error(err, { extra: { context: `redirectToCustomUrl` } });
```
Comment (leite08) [inline packages/connect-widget/src/shared/util.ts:25]: Add the `url` to `extra` as well, please, better to have as much info as possible in these

File: docs/devices-api/getting-started/connect-widget.mdx [hunk 2]
```diff
@@ -409,6 +409,22 @@ For a comprehensive list of available colors [visit here](https://www.w3schools.
   src="https://connect.metriport.com/?token=demo&customColor=orange"
 ></iframe>
 
+### redirectUrl
+
+After users successfully connect a provider, they can be redirected to a URL of your choosing.<br/>
+
+```typescript
+`https://connect.metriport.com/?token=${connectToken}&redirectUrl=https://my-app.com/good-news`;
```
Comment (Goncharo) [inline docs/devices-api/getting-started/connect-widget.mdx:417]: don't we need to encode the redirect URL before passing it as a param? ie https://www.tutorialspoint.com/how-to-encode-and-decode-url-in-typescript

so this param would be: `https%3A//my-app.com/less-good-news`

would also add a warning for the devs so they're aware

File: packages/connect-widget/src/shared/localStorage/redirect-url.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  if (url && localStorage) {
+    localStorage.setItem(Constants.SUCCESS_REDIRECT_URL_PARAM, url);
+  }
+};
+
+export const storeFailRedirectUrl = (url: string | null) => {
+  if (url && localStorage) {
+    localStorage.setItem(Constants.FAILURE_REDIRECT_URL_PARAM, url);
+  }
+};
```
Comment (Goncharo) [inline packages/connect-widget/src/shared/localStorage/redirect-url.ts:13]: ditto, need to decode URL before storing

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: docs/devices-api/getting-started/connect-widget.mdx [hunk 4]
```diff
@@ -409,6 +409,27 @@ For a comprehensive list of available colors [visit here](https://www.w3schools.
...
```
Comment (Goncharo) [inline docs/devices-api/getting-started/connect-widget.mdx:417]: the example URL should be encoded 😛

File: docs/devices-api/getting-started/connect-widget.mdx [hunk 5]
```diff
@@ -409,6 +409,27 @@ For a comprehensive list of available colors [visit here](https://www.w3schools.
...
+  For optimal results, we advise that you encode your URLs before including them in query
+  parameters. ie. redirectUrl=https://my-app.com/?query=my%20achievements
+</Tip>
+
+### failRedirectUrl
+
+After a failed attempt to connect a provider, your users can be redirected to another URL of your choosing. This is similar to [redirectUrl](#redirecturl) above.<br/>
+
+```typescript
+`https://connect.metriport.com/?token=${connectToken}&failRedirectUrl=https://my-app.com/less-good-news`;
```
Comment (Goncharo) [inline docs/devices-api/getting-started/connect-widget.mdx:430]: ditto

**Review by Goncharo — APPROVED** [review]
can 🚢 after addressing comments

**Pull request merged**